src/plugins/blocksmith/components/instagram-media-block-preview/index.jsx

```diff
@@ -1,17 +1,90 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import Message from '@gdbots/pbj/Message';
-import InstagramEmbed from 'react-instagram-embed';
 
-const InstagramMediaBlockPreview = ({ block }) => (
-  <InstagramEmbed
-    url={`https://instagr.am/p/${block.get('id')}/`}
-    hideCaption={block.get('hidecaption') || false}
-  />
-);
+export default class InstagramPostBlockPreview extends Component {
+  static propTypes = {
+    block: PropTypes.instanceOf(Message).isRequired,
+  };
 
-InstagramMediaBlockPreview.propTypes = {
-  block: PropTypes.instanceOf(Message).isRequired,
-};
+  constructor(props) {
+    super(props);
+    this.embedParentRef = React.createRef();
+    this.clean = this.clean.bind(this);
+    this.embed = this.embed.bind(this);
+    this.state = {
+      instagramID: '',
+      hideCaption: false,
+    }
+  }
 
-export default InstagramMediaBlockPreview;
+  componentDidMount() {
+    const { block } = this.props;
+    this.setState({
+      instagramID: block.get('id'),
+      hideCaption: block.get('hidecaption'),
+    }, () => {
+      const script = document.createElement('script');
```

Comment:
we may want to put this in a util, i'll let @efjacobson decide when he does a final pass on this. also my linter does not like that `instgrm` is not defined

Reply:
it looks like the others are doing it in the embed method, let's do it the same way

Reply:
nvm, spoke to eric about this one. it's fine. however let's use `window.instgrm.Embeds.process()` in both places instead

Reply:
Added window object to `instgrm.Embeds.process()`.